Server/src/main/java/org/gluu/oxauth/model/config/WebKeysConfiguration.java

```diff
@@ -6,17 +6,58 @@
 
 package org.gluu.oxauth.model.config;
 
-import javax.enterprise.inject.Vetoed;
-
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.gluu.oxauth.model.configuration.AppConfiguration;
 import org.gluu.oxauth.model.configuration.Configuration;
+import org.gluu.oxauth.model.jwk.JSONWebKey;
 import org.gluu.oxauth.model.jwk.JSONWebKeySet;
 
+import javax.enterprise.inject.Vetoed;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * @author Yuriy Movchan
  * @version 03/15/2017
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 @Vetoed
 public class WebKeysConfiguration extends JSONWebKeySet implements Configuration {
+
+    /**
+     * Configuration needed to get configuration done in LDAP about algorithms supported.
+     */
+    private AppConfiguration appConfiguration;
+
+    @Override
+    public List<JSONWebKey> getKeys() {
+        return filterKeys(super.getKeys());
+    }
+
+    /**
+     * Method responsible to filter all keys stored in LDAP with the list of
+     * algorithms that it is in Json config attribute called "jwksAlgorithmsSupported"
+     * @param allKeys All keys that are stored in LDAP
+     * @return Filtered list
+     */
+    private List<JSONWebKey> filterKeys(List<JSONWebKey> allKeys) {
```

Comment:
It works however I vote for early exit to avoid inner ifs (it's more important for big methods which we indeed should avoid).  E.g.
```
...
if (allKeys == null || allKeys.size() == 0 || jwksAlgorithmsSupported == null || jwksAlgorithmsSupported.size() == 0) {
    return allKeys;
}
...
```

Reply:
Yeah, you're right. Good sugestion.